server/handler.go

```diff
@@ -0,0 +1,109 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/benchttp/engine/internal/configparse"
+	"github.com/benchttp/engine/internal/socketio"
+	"github.com/benchttp/engine/runner"
+	"github.com/gorilla/websocket"
+)
+
+// Handler has as single method, Handler.ServeHTTP.
+// It serves a websocket server allowing remote
+// manipulation of runner.Runner.
+type Handler struct {
+	Silent   bool
+	Token    string
+	srv      *service
+	upgrader websocket.Upgrader
```

Comment:
It makes sense when passing the token from `main` as a configuration info. 

It is a cheap way of injecting the token inside the `upgrader`. 

A global variable for `token` and reading it from `upgrader` is virtually the same.

Another possibility is a closure like `newUpgrader(Handler.token)`, not really different.

Reply:
It's really not a big deal, but removing unnecessary fields makes the object slightly easier to reason about because it automatically removes unwanted dependencies.

What I was thinking about:
```go
func (h *Handler) handle(w http.ResponseWriter, r *http.Request) {
	ws, err := secureUpgrader(h.Token).Upgrade(w, r, nil)
}

func secureUpgrader(token string) *websocket.Upgrader {
	return &websocket.Upgrader{
		CheckOrigin: func(r *http.Request) bool {
			return r.URL.Query().Get("access_token") == token
		},
	}
}
```

Again, not a big deal, just making my point!